Automattic/jetpack
# Jetpack::dns_prefetch converted to using wp_resource_hints()

See #8090 and #10330

NOTE: This PR started out as #10349, looking solely at Comment Likes, then looking at all `Jetpack::dns_prefetch` functions to potentially convert to `wp_resource_hints()` and therefore deprecate the Jetpack specific function, per #8090 (comment). Now, however, it makes more sense to keep the `Jetpack::dns_prefetch` function in place, but have that pass the values over to `wp_resource_hints()` as opposed to manually managing the `<link>`'s via a custom function.

### Changes proposed in this Pull Request:
The Comment Likes feature current registers dns-prefetch resources via Jetpack's own function. Since the creation of this feature/function, wp_resource_hints() has been created, that allows a user/plugin/theme to register resource hints for use in browsers.
The `Jetpack::dns_prefetch` function is in use in approximately 6 places throughout Jetpack:

- \modules\comments.php
https://github.com/Automattic/jetpack/blob/bb884c04b32a57e3053f14c56e2f4e0138aef185/modules/comments.php#L33
- \modules\comment-likes.php
https://github.com/Automattic/jetpack/blob/bb884c04b32a57e3053f14c56e2f4e0138aef185/modules/comment-likes.php#L14
- \modules\likes.php
https://github.com/Automattic/jetpack/blob/bb884c04b32a57e3053f14c56e2f4e0138aef185/modules/likes.php#L14
- \modules\masterbar\masterbar.php
https://github.com/Automattic/jetpack/blob/bb884c04b32a57e3053f14c56e2f4e0138aef185/modules/masterbar/masterbar.php#L34
- \modules\photon-cdn.php
https://github.com/Automattic/jetpack/blob/bb884c04b32a57e3053f14c56e2f4e0138aef185/modules/photon-cdn.php#L17
- \modules\photon.php
https://github.com/Automattic/jetpack/blob/bb884c04b32a57e3053f14c56e2f4e0138aef185/modules/photon.php#L16
- \modules\shortcodes\videopress.php
https://github.com/Automattic/jetpack/blob/bb884c04b32a57e3053f14c56e2f4e0138aef185/modules/shortcodes/videopress.php#L11

###Testing instructions:

1. Enable each feature individually
2. Browse to page with that feature on
3. Check page source code with your browser's dev console. You should see <link rel="dns-prefetch" href="//url-relevant-to-feature"> is now in your head section

###Proposed changelog entry for your changes:
Utilize wp_resource_hints() function to register dns-prefetch URLs, rather than a custom function.
